include/onnxruntime/core/graph/graph.h

```diff
@@ -912,17 +990,15 @@ class Graph {
   // order if <Status> returned is "OK", otherwise it's undefined.
   common::Status PerformTopologicalSortAndCheckIsAcyclic();
 
-  common::Status PerformTypeAndShapeInferencing();
-
-  common::Status Resolve(bool no_proto_sync_required);
+  common::Status PerformTypeAndShapeInferencing(const ResolveOptions& options);
 
   // Recursively find all subgraphs including nested subgraphs
   void FindAllSubgraphs(std::vector<Graph*>& subgraphs);
 
   // Iterate this Graph instance and all subgraphs, calling the provided function for each.
   common::Status ForThisAndAllSubgraphs(const std::vector<Graph*>& subgraphs, std::function<Status(Graph&)> func);
 
-  common::Status InferAndVerifyTypeMatch(Node& node, const ONNX_NAMESPACE::OpSchema& op);
+  common::Status InferAndVerifyTypeMatch(Node& node, const ONNX_NAMESPACE::OpSchema& op, const ResolveOptions& options);
```

Comment:
@pranavsharma can you elaborate on why? as this is a private method, it seems fine to me to not have a default value and just update the existing call sites.

---
In reply to: [396856595](https://github.com/microsoft/onnxruntime/pull/3174#discussion_r396856595) [](ancestors = 396856595)

Reply:
Yeah, was trying to avoid the existing call sites. You pass the appropriately populated ResolveOptions only when you need. It's not required for all cases and is a burden for users. Up to you. not super important.

Reply:
as this PR should update existing call sites and it's not a public method, I'm inclined to leave it. thanks for the input

---
In reply to: [398843072](https://github.com/microsoft/onnxruntime/pull/3174#discussion_r398843072) [](ancestors = 398843072)